### PR TITLE
HW2 submitted by ian31174117

### PR DESCRIPTION
**Pull request by ian31174117**
---
A good PR should include the following parts:

- [x] Provide a clear title
- [x] Provide meaningful descriptions in the PR and the commit logs
- [x] Name your branch like ``<github-user-name>-hw#``, e.g., ``myname-hw2-attempt1``
- [x] Merge from your specific branch (***not master***) to the `hw*` branch (***not master***, again)
- [x] Exclude irrelevant files, such as binaries, text files, or dot files
- [x] Make sure tests/CI pass

### CODE REVIEW

File: hw2/ian31174117/q2/__pycache__/test.cpython-310-pytest-8.3.3.pyc
Comment (ThreeMonth03) [inline hw2/ian31174117/q2/__pycache__/test.cpython-310-pytest-8.3.3.pyc:1]: Please don't submit the binary files.

File: hw2/ian31174117/q2/__pycache__/test_cal_angle.cpython-310-pytest-8.3.3.pyc
Comment (ThreeMonth03) [inline hw2/ian31174117/q2/__pycache__/test_cal_angle.cpython-310-pytest-8.3.3.pyc:1]: Please remove the binary files.

**Review by ThreeMonth03 — CHANGES_REQUESTED** [review]
- [ ] Please remove the binary files.

**Comment by ian31174117** [comment]
The files have been removed, please provide a review when convenient

**Review by ThreeMonth03 — APPROVED** [review]
Thanks. Your homework looks decent, and I think it could be merged now.

**Pull request merged**